master/internal/api_token.go

```diff
@@ -0,0 +1,188 @@
+package internal
+
+import (
+	"context"
+	"fmt"
+	"time"
+
+	"github.com/uptrace/bun"
+
+	"google.golang.org/grpc/codes"
+	"google.golang.org/grpc/status"
+
+	"github.com/determined-ai/determined/master/internal/api"
+	"github.com/determined-ai/determined/master/internal/db"
+	"github.com/determined-ai/determined/master/internal/grpcutil"
+	"github.com/determined-ai/determined/master/internal/token"
+	"github.com/determined-ai/determined/master/pkg/model"
+	"github.com/determined-ai/determined/proto/pkg/apiv1"
+)
+
+// PostAccessToken takes user id and optional lifespan, description and creates an
+// access token for the given user.
+func (a *apiServer) PostAccessToken(
+	ctx context.Context, req *apiv1.PostAccessTokenRequest,
+) (*apiv1.PostAccessTokenResponse, error) {
+	curUser, _, err := grpcutil.GetUser(ctx)
+	if err != nil {
+		return nil, err
+	}
+
+	targetFullUser, err := getFullModelUser(ctx, model.UserID(req.UserId))
+	if err != nil {
+		return nil, err
+	}
+	targetUser := targetFullUser.ToUser()
```

Comment:
nit: could just use `getUser` here, and wouldn't need to call `ToUser()` on it?

Reply:
`CanCreateAccessToken` in the next few lines takes in  `"github.com/determined-ai/determined/master/pkg/model".User` type of User, and `getUser` returns `userv1.User` type. It looks like it cannot be converted to the other. 

```
	curUser, _, err := grpcutil.GetUser(ctx)
	if err != nil {
		return nil, err
	}
	targetUser, err := getUser(ctx, model.UserID(req.UserId))
	if err != nil {
		return nil, err
	}
	if err = token.AuthZProvider.Get().CanCreateAccessToken(ctx, *curUser, model.User(targetUser)); err != nil {
		return nil, status.Error(codes.PermissionDenied, err.Error())
	}
```
throws an error in `CanCreateAccessToken` line:

`cannot convert targetUser (variable of type *userv1.User) to type "github.com/determined-ai/determined/master/pkg/model".User`